HopsanGUI/Widgets/ModelWidget.cpp

```diff
@@ -175,7 +175,8 @@ ModelWidget::~ModelWidget()
     delete mpAnimationWidget;
     createOrDestroyToplevelSystem(false);
     mpSimulationThreadHandler->deleteLater();
-    mpLogDataHandler->deleteLater();
+    mpLogDataHandler->setParent(nullptr);
+    mpLogDataHandler.clear();
```

Comment:
This might be risky, clear will delete the object immediately rather then "later" (if its the last one) but this can be solved when the object is created (see other comment)